test/DynamoCoreTests/Nodes/ListTests.cs

```diff
@@ -945,7 +945,34 @@ public void LaceShortest_Simple()
 
 		}
 
-		[Test]
+        [Test]
+        public void LaceCartesian_Simple()
```

Comment:
@aparajit-pratap can we also add a ds language level imperative test to assert the for loop behavior as well?